ports/servoshell/minibrowser.rs

```diff
@@ -76,39 +117,91 @@ impl Minibrowser {
+                            }
+
+                            let location_field = ui.add_sized(
+                                ui.available_size(),
+                                egui::TextEdit::singleline(&mut *location.borrow_mut()),
+                            );
+                            if location_field.changed() {
+                                location_dirty.set(true);
+                            }
+                            if ui.input(|i| i.clone().consume_key(Modifiers::COMMAND, Key::L)) {
+                                location_field.request_focus();
+                            }
+                            if location_field.lost_focus() &&
+                                ui.input(|i| i.clone().key_pressed(Key::Enter))
+                            {
+                                event_queue.borrow_mut().push(MinibrowserEvent::Go);
+                                location_dirty.set(false);
+                            }
+                        },
+                    );
+                    ui.cursor().min.y
+                });
+            *toolbar_height = Length::new(height);
+
+            CentralPanel::default()
+                .frame(Frame::none())
+                .show(ctx, |ui| {
+                    let min = ui.cursor().min;
+                    let size = ui.available_size();
+                    let rect = egui::Rect::from_min_size(min, size);
+                    ui.allocate_space(size);
+
+                    if let Some(fbo) = servo_framebuffer_id {
+                        ui.painter().add(PaintCallback {
+                            rect,
+                            callback: Arc::new(CallbackFn::new(move |info, painter| {
+                                use glow::HasContext as _;
+                                let clip = info.viewport_in_pixels();
+                                let x = clip.left_px as gl::GLint;
+                                let y = clip.from_bottom_px as gl::GLint;
+                                let width = clip.width_px as gl::GLsizei;
+                                let height = clip.height_px as gl::GLsizei;
+                                unsafe {
+                                    painter.gl().clear_color(0.0, 0.0, 0.0, 0.0);
+                                    painter.gl().scissor(x, y, width, height);
+                                    painter.gl().enable(gl::SCISSOR_TEST);
+                                    painter.gl().clear(gl::COLOR_BUFFER_BIT);
+                                    painter.gl().disable(gl::SCISSOR_TEST);
+
+                                    let fbo = NativeFramebuffer(NonZeroU32::new(fbo).unwrap());
+                                    painter
+                                        .gl()
+                                        .bind_framebuffer(gl::READ_FRAMEBUFFER, Some(fbo));
+                                    painter.gl().bind_framebuffer(gl::DRAW_FRAMEBUFFER, None);
```

Comment:
Spending days trying to figure out why Fbo variant can't display on macOS. And it turns out CGL decided to give the default framebuffer ID with "1" :joy: 
So getting the framebuffer ID of the widget surface should fix it.

Reply:
fixed in 44ecfb6a48f931b2c087ac2e29698794fe4bbe58